src/viam/services/slam/client.py

```diff
@@ -48,10 +48,15 @@ async def get_internal_state(self, *, timeout: Optional[float] = None) -> List[b
         response: List[GetInternalStateResponse] = await self.client.GetInternalState(request, timeout=timeout)
         return [r.internal_state_chunk for r in response]
 
-    async def get_properties(self, *, timeout: Optional[float] = None) -> Tuple[bool, MappingMode.ValueType]:
+    async def get_properties(self, *, timeout: Optional[float] = None) -> Tuple[bool, MappingMode.ValueType, str, List[SensorInfo]]:
         request = GetPropertiesRequest(name=self.name)
         response: GetPropertiesResponse = await self.client.GetProperties(request, timeout=timeout)
-        return (response.cloud_slam, response.mapping_mode)
+
+        sensor_info = []
+        for si in response.sensor_info:
+            sensor_info.append(SensorInfo(name=si.name, type=si.type))
```

Comment:
Is `response.sensor_info` not already a sequence of `SensorInfo`? If so, you don't need to recast it, you could simply do `list(response.sensor_info)`